src/components/tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,266 @@
+import React, {
+  useState,
+  useRef,
+  PropsWithChildren,
+  useEffect,
+  useCallback,
+  JSXElementConstructor,
+  useMemo
+} from "react";
+import {
+  View,
+  Modal,
+  Dimensions,
+  LayoutChangeEvent,
+  TouchableWithoutFeedback
+} from "react-native";
+import { useSafeAreaInsets } from "react-native-safe-area-context";
+import { every, some } from "lodash";
+import { IOColors } from '../../core';
+import { Body, H6 } from '../typography';
+import { IconButton } from '../buttons';
+import { BottomArrow, LeftArrow, RightArrow, TopArrow } from "./Arrows";
+import {
+  ARROW_HEIGHT,
+  EMPTY_SPACE,
+  getArrowBoxByPlacement,
+  getArrowCoords,
+  getArrowVerticalAlignment,
+  getDisplayInsets,
+  getTooltipCoords,
+  getTooltipVerticalAlignment,
+  isDefined,
+  isNotZero
+} from "./utils";
+import { getChildrenPosition, tooltipStyles } from "./styles";
+import {
+  ChildrenCoords,
+  DisplayInsets,
+  Placement,
+  TooltipLayout
+} from "./utils/types";
+
+const screenDimensions = Dimensions.get("window");
+const INITIAL_COORDS: ChildrenCoords = {
+  x: 0,
+  y: 0,
+  width: 0,
+  height: 0
+};
+const ARROWS_BY_PLACEMENT: Record<
+  Placement,
+  JSXElementConstructor<{ color: string }>
+> = {
+  top: TopArrow,
+  bottom: BottomArrow,
+  left: LeftArrow,
+  right: RightArrow
+};
+
+type CommonProps = {
+  title: string;
+  content: string;
+  isVisible: boolean;
+  placement?: Placement;
+  displayInsets?: Partial<DisplayInsets>;
+  closeIconAccessibilityLabel: string;
+  childrenInteractionsEnabled?: boolean;
+  onClose: () => void;
+};
+type CloseWithTapOnBackground = {
+  allowCloseOnBackgroundTap: true;
+  backgroundAccessibilityLabel: string;
+};
+type CloseWithBackgroundTapDisabled = {
+  allowCloseOnBackgroundTap?: false;
+};
+type Props = CommonProps & (CloseWithTapOnBackground | CloseWithBackgroundTapDisabled);
+
+/**
+ * This component prompts a tooltip around its children in relation to the given `placement`.
+ * When `isVisible` is `true`
+ */
```

Comment:
I think it is better to add all the information about the props that the component accepts so that it is clear to people who use it. I have included a proposal for you but feel free to rework it

```suggestion
/**
 * Tooltip component that displays a contextual tooltip around its children.
 * The tooltip position is controlled by the `placement` prop and can adjust
 * dynamically if there is insufficient space.
 *
 * @param {React.ReactNode} children - The element around which the tooltip will be displayed.
 * @param {string} title - The title text displayed at the top of the tooltip.
 * @param {string} content - The main content text of the tooltip.
 * @param {Placement} [placement="top"] - Initial position of the tooltip; can be 'top', 'bottom', 'left', or 'right'.
 * @param {string} closeIconAccessibilityLabel - Accessibility label for the close icon button.
 * @param {boolean} isVisible - Controls the visibility of the tooltip.
 * @param {Partial<DisplayInsets>} [displayInsets={}] - Insets for adjusting tooltip position within screen boundaries.
 * @param {boolean} [allowCloseOnBackgroundTap=false] - Allows closing the tooltip by tapping outside of it.
 * @param {boolean} [childrenInteractionsEnabled=true] - Controls whether the tooltip interacts with accessibility tools.
 * @param {() => void} onClose - Callback function triggered when the tooltip is closed.
 *
 * @returns {React.ReactElement} A tooltip component rendered around the specified children.
 */
```